### ISSUE TITLE
Fix CROTA keyword in EUI maps

### ISSUE THREAD

**Issue by dstansby** [sunpy/sunpy#5274]
Currently EUI maps have a `CROTA` keyword, which by the FITS standard should really be a `CROTA2` keyword. This results in the warning
```python
/home/docs/checkouts/readthedocs.org/user_builds/solar-orbiter-python/envs/latest/lib/python3.8/site-packages/astropy/wcs/wcs.py:482: FITSFixedWarning: CROTA = 2.486914995997215 / [deg] rotation angle
keyword looks very much like CROTAn but isn't.
  wcsprm = _wcs.Wcsprm(header=header_bytes, key=key,
```
It would be good to
- Check if CROTA is in the header and CROTA2 isn't
- If so, rename the CROTA keyword to CROTA2

**Comment by wtbarnes** [comment]
Ahh I just realized that because of the way I generated the test header for the EUI source tests, the CROTA key is already replaced by the PC_ij matrix. This should be fixed too.

**Comment by ConorMacBride** [comment]
I opened a PR which adds the same code as here: https://github.com/sunpy/sunpy/blob/112dab490ef9a43078872583208210de7940d1c3/sunpy/map/sources/rhessi.py#L65-L67

Now running e.g. `sunpy.map.Map('solo_L1_eui-fsi304-image_20200303T153011632_V03.fits')` no longer produces the warning.

@wtbarnes I'm not sure I understand what needs to be fixed with the EUI source tests. Should it be using an EUI test header that only supports `CROTA2` and not a PC_ij matrix? I.e. remove `PC1_1` etc from the EUI test header?

**Comment by wtbarnes** [comment]
> Should it be using an EUI test header that only supports `CROTA2` and not a PC_ij matrix? I.e. remove `PC1_1` etc from the EUI test header?

Yep that's the idea. You *could* edit that test header directly, but I would not recommend it. The formatting is very finicky! In my opinion, your PR is fine without that being fixed.  #5261 deals with a larger refactor of the test data and should make generating these test headers much more straightforward so we can just deal with it there.

But to answer your question, the test file you mentioned used to issue the warning because it does contain `CROTA` and not the `PCi_j` (or `CROTA2`) keys. However, when I generated that test header, I passed it through the `sunpy.map.Map` factory which replaced `CROTA` with `PCi_j`, thus not making it a good test for this fix.
